basyx.aasservice/basyx.aasservice-backend-mongodb/src/main/java/org/eclipse/digitaltwin/basyx/aasservice/backend/MongoDBAasOperations.java

```diff
@@ -257,13 +240,11 @@ private List<Criteria> buildAasFilterCriteria(List<SpecificAssetId> assetIds, St
         List<Criteria> criteriaList = new ArrayList<>();
 
         // Extract globalAssetId from assetIds
-        String globalAssetId = null;
+        List<SpecificAssetId> globalAssetIds = new ArrayList<>();
         try {
-            globalAssetId = assetIds.stream()
+            globalAssetIds = assetIds.stream()
                     .filter(assetId -> "globalAssetId".equals(assetId.getName()))
-                    .findFirst()
-                    .map(SpecificAssetId::getValue)
-                    .orElse(null);
+                    .toList();
```

Comment:
Just out of curiosity - in what scenario could there be multiple `globalAssetId`'s? The only scenario I could think of is if a `globalAssetId` is set (string) and a `specificAssetId.name` is equal to `globalAssetId`. Is that the case that's handled here?